### PR TITLE
CI/CD integration for readthedocs

### PR DESCRIPTION
**Pull request by seanctech**
# Contributor Comments

* Fixed some documentation configs for readthedocs to improve layout and not pin requirements to specific version.
* Fixed issue where release task beyond 01 would not happen, also removed test task as pre-req for release. 

## Pull Request Checklist

Thank you for submitting a contribution to `easy_infra`!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:

- [x] Rebase your branch against the latest commit of the target branch
- [x] If you are adding a dependency, please explain how it was chosen
- [x] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results
- [x] If there is an issue associated with your Pull Request, align your PR
  title with [this documentation](https://help.github.com/en/articles/closing-issues-using-keywords)

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -212,11 +212,11 @@ def release(c):  # pylint: disable=unused-argument
         if pattern.fullmatch(tag.name):
             latest_release = tag.name
             break
-    else:
-        latest_release = None
+        else:
```
Comment (JonZeolla) [inline tasks.py:215]: The for else was intentional

File: tasks.py [hunk 2]
```diff
@@ -212,11 +212,11 @@ def release(c):  # pylint: disable=unused-argument
...
+            latest_release = None
 
-    if latest_release and date_info == latest_release[:7]:
-        increment = str(int(latest_release[8:]) + 1).zfill(2)
+    if date_info == latest_release[1:8]:
```
Comment (JonZeolla) [inline tasks.py:218]: This will raise an exception if latest_release is None?

**Review by JonZeolla — CHANGES_REQUESTED** [review]

**Comment by JonZeolla** [comment]
Also, to clarify, where is the actual release to rtd happening - based on a configuration in rtd that's outside of this repo watching for tags of a given regex, etc.?  I would prefer a push mechanism from the repo out to rtd, but if that is too complex then I am fine with a subscriber but we need to document the details somewhere in this repo.

File: tasks.py (hunk 1 above)
Comment (seanctech) [reply tasks.py]: Fixed.

File: tasks.py (hunk 2 above)
Comment (seanctech) [reply tasks.py]: Fixed.

**Comment by seanctech** [comment]
> Also, to clarify, where is the actual release to rtd happening - based on a configuration in rtd that's outside of this repo watching for tags of a given regex, etc.? I would prefer a push mechanism from the repo out to rtd, but if that is too complex then I am fine with a subscriber but we need to document the details somewhere in this repo.

There is a webhook in readthedocs that automatically detects when there is a new version via a CalVer regex or any merges/updates to main branch. I did find some GH Actions that can be used to trigger instead but it seems that the readthedocs webhook is the "preferred" way according to their documentation.

**Review by JonZeolla — APPROVED** [review]
LGTM

**Pull request merged**